pkg/asset/installconfig/ovirt/credentials.go

```diff
@@ -1,92 +1,231 @@
+	return nil
+}
+
+// askUsername will ask username to connect to Engine API.
+// The username provided will be added in the Config struct.
+// Returns Config and error if failure.
+func askUsername(c *Config) error {
+	err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Engine username",
+				Help:    "The username to connect to Engine API",
+				Default: "admin@internal",
+			},
+			Validate: survey.ComposeValidators(survey.Required),
 		},
-		&ovirtCertTrusted,
-		nil)
+	}, &c.Username)
 	if err != nil {
-		return c, err
+		return err
 	}
-	c.Insecure = !ovirtCertTrusted
 
-	if ovirtCertTrusted {
-		ovirtURL, err := url.Parse(c.URL)
+	return nil
+}
+
+// askCredentials will handle username and password for connecting with Engine
+func askCredentials(c Config) (Config, error) {
+	loginAttempts := 3
+	logrus.Debugf("login attempts available: %d", loginAttempts)
+	for loginAttempts > 0 {
+		err := askUsername(&c)
 		if err != nil {
-			// should have passed validation, this is unexpected
 			return c, err
 		}
-		pemURL := fmt.Sprintf(
-			"%s://%s/ovirt-engine/services/pki-resource?resource=ca-certificate&format=X509-PEM-CA",
-			ovirtURL.Scheme,
-			ovirtURL.Host)
-
-		err = survey.AskOne(&survey.Multiline{
-			Message: "oVirt certificate bundle",
-			Help:    fmt.Sprintf("The oVirt certificate bundle can be downloaded from %s.", pemURL),
-		},
-			&c.CABundle,
-			survey.ComposeValidators(survey.Required))
+
+		err = askPassword(&c)
 		if err != nil {
-			return c, err
+			loginAttempts = loginAttempts - 1
+			logrus.Debugf("login attempts now: %d", loginAttempts)
+			if loginAttempts == 0 {
+				return c, err
+			}
+		} else {
+			break
 		}
-	} else {
-		logrus.Warning("Communication with the oVirt engine will be insecure.")
 	}
+	return c, nil
+}
+
+// engineSetup will ask users: FQDN, execute validations and about
+// the credentials. In case of failure, returns Config and error
+func engineSetup() (Config, error) {
+	engineConfig := Config{}
+	httpResource := clientHTTP{}
 
-	err = survey.Ask([]*survey.Question{
+	err := survey.Ask([]*survey.Question{
 		{
 			Prompt: &survey.Input{
-				Message: "oVirt engine username",
-				Help:    "The user must have permissions to create VMs and disks on the Storage Domain with the same name as the OpenShift cluster.",
-				Default: "admin@internal",
+				Message: "Engine FQDN[:PORT]",
+				Help:    "The Engine FQDN[:PORT] (engine.example.com:443)",
 			},
 			Validate: survey.ComposeValidators(survey.Required),
 		},
-	}, &c.Username)
+	}, &engineConfig.FQDN)
 	if err != nil {
-		return c, err
+		return engineConfig, err
 	}
+	logrus.Debug("engine FQDN: ", engineConfig.FQDN)
 
-	err = survey.Ask([]*survey.Question{
-		{
-			Prompt: &survey.Password{
-				Message: "oVirt engine password",
-				Help:    "",
-			},
-			Validate: survey.ComposeValidators(survey.Required, authenticated(&c)),
-		},
-	}, &c.Password)
+	// By default, we set Insecure true
```

Comment:
-1
why is that changed?

Reply:
Because we cannot assume customer have their ssl working by default. We must check and detect first.